DependencyInjection/DoctrineExtension.php

```diff
@@ -343,6 +345,10 @@ protected function ormLoad(array $config, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
         $loader->load('orm.xml');
 
+        if (class_exists(AbstractType::class) && method_exists(DoctrineType::class, 'reset')) {
```

Comment:
I think the both parts of the condition will always be true, see https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/Form/AbstractType.php and https://github.com/symfony/symfony/blob/3.4/src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php#L280

Shouldn't the condition use `interface_exists(ResetInterface::class)`?

Reply:
I'm not sure that's sufficient: `ResetInterface` will exist when `symfony/contracts` is provided in a new-enough version. However, auto registration of the `kernel.reset` tag is dependent on `symfony/framework-bundle` 4.2 or newer. According to the changeling, we could check for the `addLink` method in `AbstractController`:
```suggestion
        // This check can be removed when support for Symfony < 4.2 is dropped
        if (method_exists(AbstractController::class, 'addLink')) {
```